src/routes/news/containers/NewsContainer.js

```diff
@@ -0,0 +1,302 @@
+|}
+
+type OwnPropsType = {| navigation: NavigationScreenProp<*>, t: TFunction |}
+type StatePropsType = StatusPropsType<ContainerPropsType, RefreshPropsType>
+type DispatchPropsType = {| dispatch: Dispatch<StoreActionType> |}
+type PropsType = {| ...OwnPropsType, ...StatePropsType, ...DispatchPropsType |}
+
+const createChangeUnavailableLanguage = (city: string, t: TFunction) => (
+  dispatch: Dispatch<StoreActionType>,
+  newLanguage: string
+) => {
+  const switchContentLanguage: SwitchContentLanguageActionType = {
+    type: 'SWITCH_CONTENT_LANGUAGE',
+    params: { newLanguage, city, t }
+  }
+  dispatch(switchContentLanguage)
+}
+
+const mapStateToProps = (
+  state: StateType,
+  ownProps: OwnPropsType
+): StatePropsType => {
+  const { t, navigation } = ownProps
+  if (!state.cityContent) {
+    return { status: 'routeNotInitialized' }
+  }
+
+  const {
+    resourceCache,
+    newsRouteMapping,
+    switchingLanguage,
+    languages
+  } = state.cityContent
+
+  const route: ?NewsRouteStateType = newsRouteMapping[navigation.state.key]
+
+  if (!route) {
+    return { status: 'routeNotInitialized' }
+  }
+  if (
+    state.cities.status === 'loading' ||
+    switchingLanguage ||
+    languages.status === 'loading'
+  ) {
+    return { status: 'loading' }
+  }
+
+  if (route.status === 'languageNotAvailable') {
+    if (languages.status === 'error') {
+      console.error(
+        'languageNotAvailable status impossible if languages not ready'
+      )
+      return {
+        status: 'error',
+        refreshProps: null,
+        code: languages.code,
+        message: languages.message
+      }
+    }
+    return {
+      status: 'languageNotAvailable',
+      availableLanguages: languages.models.filter(lng =>
+        route.allAvailableLanguages.has(lng.code)
+      ),
+      changeUnavailableLanguage: createChangeUnavailableLanguage(route.city, t),
+      innerProps: {
+        language: state.contentLanguage,
+        cityCode: route.city,
+        selectedNewsType: route.type,
+        navigation: ownProps.navigation
+      }
+    }
+  }
+
+  const refreshProps = {
+    path: route.path,
+    cityCode: route.city,
+    language: state.contentLanguage,
+    navigation: ownProps.navigation,
+    selectedNewsType: route.type
+  }
+
+  if (state.cities.status === 'error') {
+    return {
+      status: 'error',
+      message: state.cities.message,
+      code: state.cities.code,
+      refreshProps
+    }
+  } else if (resourceCache.status === 'error') {
+    return {
+      status: 'error',
+      message: resourceCache.message,
+      code: resourceCache.code,
+      refreshProps
+    }
+  } else if (route.status === 'error') {
+    return {
+      status: 'error',
+      message: route.message,
+      code: route.code,
+      refreshProps
+    }
+  } else if (languages.status === 'error') {
+    return {
+      status: 'error',
+      message: languages.message,
+      code: languages.code,
+      refreshProps
+    }
+  }
```

Comment:
Cant we do something like this? Or does flow prevent this?
```suggestion
  if (state.cities.status === 'error' || resourceCache.status === 'error' || ...) {
    return {
      status: 'error',
      message: state.cities.message || resourceCache.message || ...,
      code: state.cities.code || resourceCache.code || ...,
      refreshProps
    }
  }
```

Reply:
I remember flow prevents it. But let me re-check this part.

Reply:
yes it's not possible to re-factor it this way because flow type check fails